### PR TITLE
Remove the interface{} from being a valid argument when deep down go-kit can error out.

### PR DESCRIPTION
**Pull request by InfernoJJ**
Remove the interface{} from being a valid argument when deep down go-kit can error out.

### CODE REVIEW

File: log/model_formats.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+func Timestamp(t time.Time) Renderer {
+	return TimestampFormat(t, time.RFC3339Nano)
+}
+
+func TimestampFormat(t time.Time, format string) Renderer {
+	return String(t.Format(format))
+}
+
+func ByteString(b []byte) Renderer {
```
Comment (vxio) [inline log/model_formats.go:51]:
```suggestion
func Bytes(b []byte) Renderer {
```

File: log/model_formats.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+func Float64(f float64) Renderer {
+	return &any{f}
+}
+
+func Bool(b bool) Renderer {
+	return &any{b}
+}
+
+func Duration(d time.Duration) Renderer {
```
Comment (vxio) [inline log/model_formats.go:39]:
```suggestion
func TimeDuration(d time.Duration) Renderer {
```

This kind of makes it nice for autocompletion purposes.

File: log/model_formats.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	return &any{d.String()}
+}
+
+func Timestamp(t time.Time) Renderer {
```
Comment (vxio) [inline log/model_formats.go:43]:
```suggestion
func Time(t time.Time) Renderer {
```

File: log/model_formats.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	return TimestampFormat(t, time.RFC3339Nano)
+}
+
+func TimestampFormat(t time.Time, format string) Renderer {
```
Comment (vxio) [inline log/model_formats.go:47]:
```suggestion
func TimeFormatted(t time.Time, format string) Renderer {
```

File: log/model_formats.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+import (
+	"encoding/base64"
+	"time"
+)
+
+// Renderer is an interface to deal with typing problems of just having an interface{} as the acceptable parameters
+// Go-kit logging has a failure case if you attempt to throw any values into it.
+// This is a way to guard our developers from having to worry about error cases of the lower logging framework.
+type Renderer interface {
```
Comment (vxio) [inline log/model_formats.go:11]: nit: should we call this `Valuer`?

**Comment by vxio** [comment]
Could we add in method chaining? 

```go
// current
log = log.Debug()
log.Set("ts", log.Time(time.Now().UTC()))
log.Set("foo", log.String("bar"))

// version with method chaining
log.Debug().
      Time("ts", time.Now().UTC()).
      String("foo","bar")
     
```

File: log/model_formats.go (hunk 1 above)
Comment (InfernoJJ) [reply log/model_formats.go:51]: I think it makes sense that we're letting them know were casting it to a string.

**Review by adamdecaf — APPROVED** [review]

**Review by vxio — APPROVED** [review]

**Pull request merged**